### DISCUSSION TITLE
String Replacement and combining templates

### DISCUSSION THREAD

**Discussion by dredhorse** [SilentVoid13/Templater#630] [Help]
Hi, something more complicated and I don't know how to start.

a) I would like to have a standard frontmatter in all of my notes. The frontmatter should contain some Token which should become replaced when I create the note.
b) depending on location or filename a different template should be used.

So for example I have a frontmatter of:

---
UUID: <% tp.date.now("YYYYMMDDHHmm") %>
DateStamp: <% tp.date.now("YYYY-MM-DD") %>
Type: {{Type}}
Status: {{Status}}
---

and then a template for a Meeting

This Date
Attendees etc...

The endresult should be:

---
UUID: <% tp.date.now("YYYYMMDDHHmm") %>
DateStamp: <% tp.date.now("YYYY-MM-DD") %>
Type: Meeting
Status: Prepare
---
This Date
Attendees etc...

I already found a way to do something based on the filename but I don't know how I can merge several templates, replace the tokens in them and then insert that information into the note.

Any Idea? Or should I just copy / paste the yaml frontmatter in every template, which would only require me to replace the token and then insert the information into the note?

**Comment by AB1908** [comment]
Look at `tp.file.include`

**Reply by dredhorse** [reply]
ok that would include the content of the file, but how do I replace the tokens in it? I don't want to replace the token in the file directly only in the note I'm in.

**Reply by JeromeZephyr** [reply]
The `tp.file.include` is useful to reuse part of your template. See: [tp.file - Templater](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfileincludeinclude_link-string--tfile)

I would use this for your two first lines. 
The rest could be dynamic depending on your directory. You can setup the default templater in function on where the note is created.

You could also do conditions using the [tp.file - Templater](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilepathrelative-boolean--false). Exemples: [Whitespace Control - Templater](https://silentvoid13.github.io/Templater/commands/whitespace-control.html)

**Comment by dredhorse** [comment]
I think I found a way, thanks for helping.

**Reply by AB1908** [reply]
Can you describe your solution here for posterity?

**Comment by dredhorse** [comment]
I will after I figured out how it finally should work. I need to probably post it as a gist as the editor here breaks the formatting.
The main concept atm is:
NewNote Template:
- Asks for some stuff
- Sets Title
- includes default Frontmatter without an ending ---
- includes Template for kind of note
Specific Type Template:
- adds missing Frontmatter and closes it with ---
- does the rest

I'm also looking into other approaches, like removing "own" metadata from Frontmatter into an own metadata and editing the frontmatter on the fly afterwards.

While both other avenues work the templater plugin causes issues. It is probably my lack of understanding but I think it is based on how the include function and other plugins and obsidian functions work.

To me it looks like that the note is only fully saved after the full script has run, which is a problem if you want to access the frontmatter at the end of the script.

I think one missing feature of Templater is a way to trigger a template after a certain amount of "runs" like <%+ should do... but I can't get that to work and it requires switching between modes.

I'm looking into it.

**Reply by huyz** [reply]
Posterity awaits :)
